src/EventSourcing/Domain/Aggregate/AggregateChanged.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EventSourcingWorkshop\EventSourcing\Domain\Aggregate;
+
+/**
+ * Represents a set of {@see AggregateDomainEvent} produced when interacting with an {@see Aggregate}:
+ * this is designed to work well with {@see AggregateRepository}, so it is easy to save the changes that
+ * occurred within an {@see Aggregate}.
+ *
+ * @psalm-immutable
+ * @psalm-template AggregateType of Aggregate
+ */
+final class AggregateChanged
+{
+    public AggregateId $aggregate;
```

Comment:
These can be `readonly`. In fact, a sweep for `readonly` properties should be done.

Reply:
I need to do a general hunt for `readonly` properties

Reply:
014148f32228de70d2f12b930accb356cfe0f07e